File: Daily_Challenge/20230202_LC_6.py
# Date: 2023.02.02
# Written by Jian-Yuan Lin

class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1 or numRows >= len(s):
            return s
        word = [""] * numRows
        step, idx, i = 1, 0, 0
        while i < len(s):
            word[idx] += s[i]
            idx += step
            if idx == 0:
                step = 1
            elif idx == numRows - 1:
                step = -1
            i += 1
        ans = ""
        for s in word:
            ans += s
        return ans

File: Daily_Challenge/20230202_LC_6.cpp
// Date: 2023.02.02
// Written by Jian-Yuan Lin

class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows == 1 || numRows >= s.size()){
            return s;
        }
        vector<string> word(numRows, "");
        int step = 1, idx = 0, i = 0;
        while (i < s.size()){
            word[idx] += s[i];
            idx += step;
            if (idx == 0){
                step = 1;
            }
            else if (idx == numRows - 1){
                step = -1;
            }
            i++;
        }
        string ans = "";
        for (string s: word){
            ans += s;
        }
        return ans;
    }
};